vllm/worker/habana_model_runner.py

```diff
@@ -1018,34 +1046,78 @@ def create_dummy_seq_group_metadata(self, group_id, seq_len, is_prompt):
         output_token_ids = [1] * output_len
         seq_data = SequenceData(prompt_token_ids)
         seq_data.output_token_ids = output_token_ids
-        return SequenceGroupMetadata(
-            request_id=str(group_id),
-            is_prompt=(output_len == 0),
-            seq_data={group_id: seq_data},
-            sampling_params=sampling_params,
-            block_tables=block_tables,
-        )
+        return SequenceGroupMetadata(request_id=str(group_id),
+                                     is_prompt=(output_len == 0),
+                                     seq_data={group_id: seq_data},
+                                     sampling_params=sampling_params,
+                                     block_tables=block_tables,
+                                     lora_request=lora_request)
 
     def profile_run(self) -> None:
         num_layers = self.model_config.get_num_layers(self.parallel_config)
         kv_caches = [None] * num_layers
         max_batch_size = self.prompt_bs_bucket_cfg[-1]
         max_seq_len = self.prompt_seq_bucket_cfg[-1]
-
-        self.warmup_scenario(max_batch_size, max_seq_len, True, kv_caches)
-
-    def warmup_scenario(self, batch_size, seq_len, is_prompt,
-                        kv_caches) -> None:
+        if self.lora_config:
+            max_seq_len = self.max_num_batched_tokens // max_batch_size
```

Comment:
Why overwrite number in L1060? Should this be a min of the 2?

Reply:
This is overwritten to ensure that max_seq_len * match_batch_size does not exceed max_num_batched_tokens at any point of time. As the Lora layers can handle only up to that.